src/main/groovy/com/ullink/SourceLinkIndexing.groovy

```diff
@@ -42,6 +42,10 @@ class SourceLinkIndexing extends ConventionTask {
         }
     }
 
+    def findRepoRootDir(File dir) {
+        new File((File) dir, '.git').exists() ? dir : findRepoRootDir(dir.getParentFile())
```

Comment:
Well, the navigation upwards the tree shouldn't be that much of a problem no? I mean, there's a finite steps it takes to reach root.